lego/apps/events/serializers/events.py

```diff
@@ -79,6 +80,27 @@ class Meta:
         )
         read_only = True
 
+    def user_should_see_regs(self, event, user):
+        return event.get_possible_pools(user, future=True, is_admitted=False).exists() or \
+               user.is_abakom_member or \
+               event.created_by.id == user.id
+
+    def get_pools(self, obj):
+        request = self.context.get('request', None)
+        queryset = obj.pools.all()
+        if request and request.user.is_authenticated:
+            self.context['should_see_regs'] = self.user_should_see_regs(obj, request.user)
+        return PoolReadSerializer(queryset, context=self.context, many=True).data
+
+    def get_waiting_registrations(self, obj):
+        request = self.context.get('request', None)
+        if request and \
+                request.user.is_authenticated and \
+                self.user_should_see_regs(obj, request.user):
+            queryset = obj.waiting_registrations
+            return RegistrationReadSerializer(queryset, context=self.context, many=True).data
+        return obj.waiting_registrations.count()
```

Comment:
Is it possible to return this count as `registrationCount` instead? Same for waiting list..

Would make typings in the frontend way better.. :smile:

Reply:
Yep, just add `registration_count` as we do [here](https://github.com/webkom/lego/blob/ccd839e9ee61508f49afff487df7259cfdd029ab/lego/apps/events/serializers/events.py#L54) to the detailed serializer @edvardvb

Reply:
And drop these `.count()` here.

Reply:
Hmm. If you remove count here, it will return all the registrations for unauthorized users.

It is still nice to get pools with registration counts, not just the total count.

Reply:
SerializerMethodField does not return anything implicitly, does it?

Right, I was only thinking of the total - like we use for the calendar. We should probably support showing the pools with separate counters on the event page. Could add a registrationCount property, similar to the one for event, for the pools and add it in the pool serializer.

Reply:
Oh, I see what you meant. I meant to drop the whole line ofc :)